src/components/validation/resources/http.rs

```diff
@@ -257,13 +267,26 @@ fn spawn_output_http_server(
     // First, we'll build and spawn our HTTP server.
     let decoder = codec.into_decoder()?;
 
+    // the number of events we expect to receive from the sink.
+    // any happy path events + any events that are flagged as external resource should reject.
+    let expected_output_events = input_events
+        .iter()
+        .filter(|te| !te.should_fail() || te.should_reject())
+        .count();
+
+    let should_reject = input_events.iter().filter(|te| te.should_reject()).count() > 0;
```

Comment:
One thing that jumps out to me here is that the logic, even though it may appear to, doesn't actually allow for differentiation between good and bad events. All events would be rejected even if only one of the input events is actually marked to cause a rejection.

This is another comment that isn't meant to indicate that I think you need to fix/change this, but just something worth thinking about.

Reply:
Right, this did cross my mind as well. It is worth a code comment, I'll add one. The reason I didn't go down that path was because the test cases so far currently don't require that functionality, but we certainly might need it. And the way to solve it would be a bit tricky because we can't guarantee the order of events received (I believe). So we'd have to have some kind of tagging mechanism in the test case events that trickles down into here to say "if you receive an event with this field, reject it".